relay-cardinality/src/limiter.rs

```diff
@@ -172,14 +215,16 @@ pub struct CardinalityLimits<'a, T> {
     source: Vec<T>,
     rejections: HashSet<usize>,
     limits: HashSet<&'a CardinalityLimit>,
+    reports: BTreeMap<&'a CardinalityLimit, Vec<CardinalityReport>>,
```

Comment:
Could this be a single map instead of a hash set and a btreemap, with an empty `Vec` if `limit.report` is false?

Reply:
I just implemented it, but I it's actually wrong. There can be cardinality reports for limits which have not been exceeded.

I'll revert and rename the field to make it more clear!